### ISSUE TITLE
Will Clipy be updated for M1 processors?

### ISSUE THREAD

**Issue by mikemorgenstern** [Clipy/Clipy#430]
Hi all! First off, thank you for writing the single most useful program I use on my computer.

I'm considering upgrading to an M1 computer and wondering if you'll be releasing a universal binary for Clipy?

Thanks very much!

**Comment by MalEbenSo** [comment]
+1

**Comment by Econa77** [comment]
It will be supported. The M1 Mac will be delivered to me by the end of this year, so please wait a little longer.

**Comment by MalEbenSo** [comment]
Thanks @Econa77. 

Off-topic #1: I have received my M1 laptop and I love it. Well, it's just a laptop after all. But so much of it just feels right. And I already notice that my battery anxiety has dropped significantly. I pay less and less attention to the symbol in the menubar. 

Off-topic #2: Not to push but out of curiosity as a non-developer. I understood that one can develop and compile for either platform (M1, x86) from either platform. Is that not so? (because you are waiting for your M1 Mac).

**Comment by zhixiangyao** [comment]
> It will be supported. The M1 Mac will be delivered to me by the end of this year, so please wait a little longer.

nice!  ありがとう！

**Comment by teplynin** [comment]
Any news about version for Apple M1?

**Comment by liiluu** [comment]
Oddly enough Clipy was working for me on my M1 chip, though slightly buggy, it would crash whatever program I was using if I wanted to paste an image that was in anything other than the first clipboard spot. So I deleted it, in hopes of downloading a newer version, only to have the issue that the program downloads fine but won't open. Despite looking like it's opening the application, it doesn't show in my toolbar nor in any of my open apps.

**Comment by mntbighker** [comment]
I have been using Clippy on my M1 mini without much issue. It may well be that my usage habits don't expose the issues or bugs others may be seeing. I will say that Clippy is ESSENTIAL to my job though. I would be back on an Intel Mac if Clippy were not working for me. I'm looking forward to an M1 native version.

**Comment by dromeroDev** [comment]
M1 native version please +1

**Comment by rene-dohan** [comment]
https://github.com/rene-dohan/Clipy/releases/tag/2.0.0

**Comment by dromeroDev** [comment]
[Crippled version that is working native on apple silicon](https://github.com/rene-dohan/Clipy/releases/tag/2.0.0)

**Comment by houcros** [comment]
> [Crippled version that is working native on apple silicon](https://github.com/rene-dohan/Clipy/releases/tag/2.0.0)

Broken link. Seems that fork has disappeared.

**Comment by stevenwenxu** [comment]
Hello @Econa77, thanks for making this great tool that I have been using for over 3 years. Since you [already fixed](https://github.com/Clipy/Clipy/pull/465) building for m1, could you please make an official release?

**Comment by mntbighker** [comment]
Looking forward to native M1 support here too. Not that I have particular issues with it now.

**Comment by foliovision** [comment]
I'm waiting on a native M1 version. Having Rosetta 2 running in the background uses up memory and somehow slows down an M1 Mac. Tried first with Mac Mini 8GB RAM and later with an M1 Pro MacBook Pro 14 with 16GB RAM. I'm not really a Clipy refugee but a ClipMenu user who is moving straight from Mojave to Monterey (with a brief visit of Big Sur on 2009 Intel Mac Pros and M1 Mac Mini). None of the alternatives compare with efficient ClipMenu and Clipy with too much interface, inadequate keyboard shortcuts and way too much code.

**Comment by razakhawaja** [comment]
What's the update on this?

**Comment by gagarine** [comment]
>  Having Rosetta 2 running in the background

@mntbighker Rosetta 2 does not run in the background. It translates "intel code" (x86-64) to "M1 code" the first time the app run (I believe it also support just in time compilation but seem to use ahead of time compilation most of the time). Furthermore, rosetta 2 complied code is fast mainly because Apple Silicon support of x86-64 memory ordering.

In practice, on my M1 Clipy take about 30Mb ram. With 16GB it should definitely be no issue. When not pasting something Clipy is using less than 0.1% on my M1 air... so again should not be an issue. It take about 5% when pasting, and this could eventually be improved but really... who care?

For sure native M1 support would be a nice to have, but if your computer is slow it as certainly nothing to do with Clipy. Or it's a bug.

**Comment by mlcampbe** [comment]
Is it possible that we can download and recompile the source on an M1 machine to get a native version? 

Update: I tried and it failed to compile. Getting several warnings which I can ignore but some of the files just won't compile. But, I did find a pre-compiled version at https://github.com/ian4hu/Clipy that is a native ARM release.

**Comment by foliovision** [comment]
> But, I did find a pre-compiled version at https://github.com/ian4hu/Clipy that is a native ARM release.

Yes, good find. Thanks to your pointer, I'm now using the ian4hu ARM version on an M1 Max on Monterey and it works a treat. Great to have Clipy back.

I've also installed his Universal Binary on an Intel Mac Mini on Big Sur and it works fine there too. There's quite a few releases between the main Clipy version and the ian4hu versions since then. It would be great if the two teams could somehow join forces again.

**Comment by MalEbenSo** [comment]
Joining efforts would be helpful for other reasons, too:
Building for different architectures can improve code quality. 
And so that more people look at the code – a clipboard manager is quite sensitive to data privacy.

**Comment by yourfavorite** [comment]
Normally I wouldn't point to another project like this, but since clipy isn't really being maintained anymore... For those that want very similar functionality on a project that is still being maintained, I'd recommend checking out https://maccy.app

**Comment by foliovision** [comment]
I'm not normally a fan of pointing to other projects either. [Ian Hu's Clipy](https://github.com/ian4hu/Clipy) is working great on my M1 Max. But out of foolish curiousity, I am trying Maccy. Maccy installs okay [from Github](https://github.com/p0deje/Maccy) and does seem to do what it promises in a minimalist way. There is a release-quality [M1 version](https://github.com/p0deje/Maccy/issues/199) since January 2021.

I.e. this is not a long term review but Maccy is worth checking out and worth donating to via Gumroad is it suits some better than Clipy. The search is a neat addition.

**Comment by fasfsfgs** [comment]
First of all, I'm such a huge fan of Clipy! It's in my laptop for several years now.
I just received my new M1 macbook today and the first thing I did was to look for arm64's version on Clipy, but sadly I found out that there is none "official" at the moment.

So I'm using Maccy, mentioned by @yourfavorite and @foliovision, and I have to say it seems really really great.

**Comment by arossert** [comment]
Is there any good alternative that has the same simplicity and snippets feature as Clipy?
I'm using [this](https://github.com/harryzjm/Clipy) fork that does work on my M1 mac but for some reason, I lost the ability to edit the snippets, they do exist from my previous mac before I migrated to the M1.
I try several other forks like [this](https://github.com/ian4hu/Clipy) but they are all crashing and do not even start

[Removed trace to not offend anyone]

**Comment by AlecKinnear** [comment]
Did you take the time to try the version of Clipy which I have working (noted above) from Ian Hu and which is still working fine for me on 12.6.1 (no Ventura going anywhere my M1 MacBook Pro for another year or two).

Or perhaps you took the time to try Maccy?

Or @arossert did you just start by posting a huge error report on a different version of Clipy in the wrong project to clutter up what was a very useful thread and vandalise Github?

**Comment by arossert** [comment]
@AlecKinnear You probably did not read my post if you ask these questions, I did try this fork and this is the error that I'm getting since I did not get an answer on that fork repo I tried here.

I looked at Maccy as mentioned in this thread but I do love the simplicity of Clipy and the ability to use snippets.

I hoped to get some direction on how to get Ian Hu fork to work but I see that you got very offended by my post so I will remove the error so you can be in peace. 

Thanks for trying to help.

**Comment by AlecKinnear** [comment]
@arossert The [Ian Hu M1 Clipy](https://github.com/ian4hu/Clipy) is working without error here. I'm sorry you are having issues. Maccy is not as simple as Clipy, I agree, but it's a very good crutch in the mean time. I was able to be productive with Maccy for about six weeks while I trialed Maccy as my primary multiclipboard.

Hopefully the main Clipy developers will look at what Ian Hu has done and create an official version with full testing sometime soon.

**Comment by arossert** [comment]
Seems like the last release was at 2018 so I will not get my hopes high ☺️

**Comment by mntbighker** [comment]
I have switched to ClipTools in the App Store

**Comment by davydof** [comment]
@Econa77 hi, any update? :)

**Comment by AlecKinnear** [comment]
> ClipTools in the App Store

[Looks promising](https://apps.apple.com/ca/app/cliptools/id1619348240?mt=12). And it's free. I'm still enjoying [Ian Hu's M1 Clipy](https://github.com/ian4hu/Clipy) on Monterey 12.6.3 so I won't switch, but for anyone having issues with M1 Clipy, definitely worth a look.

**Comment by davydof** [comment]
IDK how but version 1.2.1 works
https://clipy-app.com/

```
MacBook Pro M1 Pro
Ventura 13.2.1
```

**Comment by MalEbenSo** [comment]
I have switched to Maccy, which works well for me.

**Comment by dievilz** [comment]
I switched to Maccy too, works nice!

**Comment by lizy14** [comment]
I'm on M1 chip macOS 26.4.1, Clipy 1.2.1 has been working well. I recently started to see pop-ups from Apple that says Intel support will be dropped soon. Does it mean I have to switch to ian4hu's fork?

**Comment by Econa77** [comment]
@lizy14 Thank you for the comment, and sorry for the concern caused by the long period without an official release.

As mentioned in #590, I have restarted development and maintenance of Clipy. I’m currently working on the remaining tasks needed for the next official release, including Apple Silicon support.

My plan is to release an Apple Silicon compatible official version before the next macOS release. Please wait a little longer while I finish the remaining work.

**Comment by Econa77** [comment]
As mentioned in https://github.com/Clipy/Clipy/issues/590, I have published a draft release of Clipy 1.2.2:

https://github.com/Clipy/Clipy/releases/tag/1.2.2

This build is a Universal Binary and supports both Intel and Apple Silicon.

I am currently collecting feedback before preparing the official v1.3.0 release. If you are willing to help test it, please install 1.2.2 and let me know if you notice any issues, migration problems, or unusual behavior.

Before updating, please export any important data, including snippets, just in case. When reporting feedback, it would be very helpful if you could include your macOS version and Mac model.

**Comment by Econa77** [comment]
[v1.3.0](https://github.com/Clipy/Clipy/releases/tag/1.3.0) has now been released and rolled out 🎉 

Thank you again to everyone who has supported Clipy for such a long time, especially during the long period when development was inactive. Your feedback, testing, reports, pull requests, forks, and encouragement helped make this restart possible.

I will continue maintaining and improving Clipy going forward, and I would appreciate your continued help.

If you notice any issues after updating to v1.3.0, please open a new issue with the details.

Thank you again for your support 👍
